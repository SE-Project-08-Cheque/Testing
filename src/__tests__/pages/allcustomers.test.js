import React from 'react'
import {render, screen , waitFor} from '@testing-library/react'
import AllCustomers from '../../pages/allcustomers'
import { BrowserRouter } from "react-router-dom";
import {mocked} from 'ts-jest/utils'
import renderer from 'react-test-renderer'
import API_Service from '../../Services/API_Service';

jest.mock('../../Services/API_Service');
const mockgetAllDataCustomerTable = mocked(API_Service.getAllDataCustomerTable,true)
console.log(mockgetAllDataCustomerTable)

describe('When evrything is OK in allcustomers page',()=>{
    beforeEach(async()=>{
        render(<BrowserRouter><AllCustomers/></BrowserRouter>)
        await waitFor(()=>expect(mockgetAllDataCustomerTable).toHaveBeenCalled())
    })
    test('should render the AllCustomer Page without crashing',()=>{
        screen.debug()
    })
    test('should render the AllCustomer Page without crashing',()=>{
        screen.getByText("Customer ID")
        screen.getByText("Account No")
        screen.getByText("NIC")
        screen.getByText("Contact")
        screen.getByText("Created Date")
        screen.getByText("Postal Code")
    })
})

describe("when the component fethches the details successfully",()=>{
    beforeEach(()=>{
        mockgetAllDataCustomerTable.mockClear();
    });
    test('should call Allcustomers once',async ()=>{
        render(<BrowserRouter><AllCustomers/></BrowserRouter>)
        await waitFor(()=>expect(mockgetAllDataCustomerTable).toHaveBeenCalledTimes(1))
    })
    test('should renders allcustomers details correctly',async ()=>{
        mockgetAllDataCustomerTable.mockImplementation((callback)=>{
            callback({data:{data:[
            {NIC: "766855383V", account_no: 1234567892, city: "negombo", contact_primary: 728908140, created_date: "2020-02-10", customer_id: 33, dob: "2001-02-22", email: "dakshitha@gmail.com", full_name: "Dakshitha Suriyaaratchie", gender: "Male",
            house_no: "123/2", name_with_init: "name_with_ini1", pin: 133456, postal_code: 11320000, street: "streee"}]}})
          })
    render(<BrowserRouter><AllCustomers/></BrowserRouter>)
    screen.getByText('766855383V')
    screen.getByText('1234567892')
    screen.getByText('negombo')
    screen.getByText('728908140')
    screen.getByText('2020-02-10')
    screen.getByText('33')
    screen.getByText('2001-02-22')
    screen.getByText('dakshitha@gmail.com')
    screen.getByText('Dakshitha Suriyaaratchie')
    screen.getByText('Male')
    // screen.debug()
    })
    test("matches allcustomers snapshot",()=>{
        const tree=renderer.create(<AllCustomers/>).toJSON()
        expect(tree).toMatchSnapshot();
    })
})