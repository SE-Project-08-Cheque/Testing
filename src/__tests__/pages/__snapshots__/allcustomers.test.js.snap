// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the component fethches the details successfully matches allcustomers snapshot 1`] = `
<div
  className="makeStyles-divi-34"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": "50px",
      }
    }
  >
    <div
      className="e-card e-card-horizontal"
      style={
        Object {
          "width": "400px",
        }
      }
    >
      <img
        alt="Sample"
        src=""
        style={
          Object {
            "height": "80px",
          }
        }
      />
      <div
        className="e-card-stacked"
      >
        <div
          className="e-card-header"
        >
          <div
            className="e-card-header-caption"
          />
        </div>
        <div
          className="e-card-content"
        />
      </div>
    </div>
  </div>
  <div
    className="css-gmuwbf"
  >
    <div
      className="MuiPaper-root makeStyles-root-32 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiTableContainer-root makeStyles-container-33"
      >
        <table
          aria-label="sticky table"
          className="MuiTable-root MuiTable-stickyHeader"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-35 MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
              >
                Customer ID
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Account No
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                NIC
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Full Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                City
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Postal Code
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Gender
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Email
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Date of Birth
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Created Date
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
              >
                Contact
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            className="MuiTablePagination-spacer"
          />
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              10
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={10}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            0-0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Previous page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Next page"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
